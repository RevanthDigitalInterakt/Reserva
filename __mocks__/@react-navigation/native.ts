const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
