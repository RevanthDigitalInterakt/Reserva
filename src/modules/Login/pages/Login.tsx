import { useMutation } from '@apollo/client';
import AsyncStorage from '@react-native-community/async-storage';
import { useNavigation } from '@react-navigation/native';
import { StackScreenProps } from '@react-navigation/stack';
import * as React from 'react';
import { useEffect, useState } from 'react';
import { BackHandler, SafeAreaView, ScrollView } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { Box, Button, Typography } from 'reserva-ui';
import { images } from '../../../assets';
import {
  classicSignInMutation,
  sendEmailVerificationMutation,
} from '../../../graphql/login/loginMutations';
import { useAuth } from '../../../context/AuthContext';
import { RootStackParamList } from '../../../routes/StackNavigator';
import HeaderBanner from '../../Forgot/componet/HeaderBanner';
import UnderlineInput from '../components/UnderlineInput';
// import { string } from "yup";
import * as Yup from "yup";
type Props = StackScreenProps<RootStackParamList, 'LoginAlternative'>;

export const LoginScreen: React.FC<Props> = ({
  children,
  route,
  navigation,
}) => {
  const { comeFrom } = route.params;
  const { cookie, setCookie, setEmail } = useAuth();
  //const navigation = useNavigation();
  const [loginCredentials, setLoginCredentials] = React.useState({
    username: '',
    password: '',
    showPasswordError: false,
    passwordError: '',
    showUsernameError: false,
    usernameError: '',
    hasError: false,
    showMessageError: '',

  });
  const [isSecureText, setIsSecureText] = useState(true);
  const [showError, setShowError] = useState(false);
  const [emailIsValid, setEmailIsValid] = useState(false);
  const [passwordIsValid, setPasswordIsValid] = useState(false);
  const [login, { data, loading }] = useMutation(classicSignInMutation);
  const [loginWithCode, setLoginWithCode] = useState(true);

  const [sendEmail, { loading: loadingSendMail, data: dataSendMail }] =
    useMutation(sendEmailVerificationMutation);

  const validateCredentials = () => {
    setLoginCredentials({
      ...loginCredentials,
      showPasswordError: true,
      showUsernameError: true,
      hasError: true,
      showMessageError: 'Verifique os campos acima e digite um e-mail ou senha válidos'
    });
  }

  const removeMessageErrorEmail = () => {
    setLoginCredentials({
      ...loginCredentials,
      showUsernameError: false,
      usernameError: ''
    });
  }
    ;
  const handleLogin = async () => {
    if (emailIsValid && passwordIsValid) {
      const { data, errors } = await login({
        variables: {
          email: loginCredentials.username,
          password: loginCredentials.password,
        },
      });
      if (data['classicSignIn'] === 'Success') {
        setEmail(loginCredentials.username)
        navigation.navigate('Home');
      } else {
        validateCredentials();
      }
    } else {
      validateCredentials();
    }

  };

  const handleLoginCode = () => {
    if (emailIsValid) {
      removeMessageErrorEmail();
      sendEmail({
        variables: {
          email: loginCredentials.username,
        },
      }).then((data) => {
        setEmail(loginCredentials.username)
        navigation.navigate('AccessCode', {
          email: loginCredentials.username,
        });
      });
    } else {
      setLoginCredentials({
        ...loginCredentials,
        showUsernameError: true,
        usernameError: 'Digite um e-mail válido'
      });
    }

  };

  useEffect(() => {
    if (comeFrom === 'Profile') {
      BackHandler.addEventListener('hardwareBackPress', () => {
        navigation.navigate('Home');
        return true;
      });
    }
  }, []);

  useEffect(() => {
    if (!loading && data?.cookie) {
      console.log('cookie', data.cookie);
      setCookie(data?.cookie);
      AsyncStorage.setItem('@RNAuth:cookie', data?.cookie).then(() => {
        navigation.navigate('Home');
      });
    }
  }, [data]);

  return (
    <SafeAreaView style={{ backgroundColor: 'white' }} flex={1}>
      <HeaderBanner
        imageHeader={images.headerLogin}
        onClickGoBack={() => {
          if (comeFrom === 'Profile') {
            navigation.navigate('Home');
          } else {
            navigation.goBack()
          }
        }}
      />
      <ScrollView>
        <Box px="xxs" pt="xxs" paddingBottom="xxl">
          <Typography fontFamily="reservaSerifRegular" fontSize={22}>
            Seja bem-vindo!
          </Typography>

          <Box mt="xxs">
            <Box marginBottom="xxxs">
              <Typography variant="tituloSessao">
                Insira seu e-mail para continuar:
              </Typography>
            </Box>
            <UnderlineInput
              placeholder="Digite seu e-mail"
              keyboardType="email-address"
              isSecureText={false}
              value={loginCredentials.username}
              showError={loginCredentials.showUsernameError}
              errorMsg={loginCredentials.usernameError}
              onChangeText={(text) => {
                setLoginCredentials({ ...loginCredentials, username: text })
                setEmailIsValid(
                  Yup.string()
                    .required()
                    .email()
                    .isValidSync(text)
                );
              }
              }
            />

            {!loginWithCode && (
              <Box mt="md" width="100%">
                <UnderlineInput
                  isSecureText={true}
                  placeholder="Digite sua senha"
                  value={loginCredentials.password}
                  showError={loginCredentials.showPasswordError}
                  onChangeText={(text) => {
                    setLoginCredentials({ ...loginCredentials, password: text })
                    setPasswordIsValid(
                      Yup.string()
                        .required()
                        .matches(/^(?=.{8,})/) // 8 caracteres
                        .matches(/^(?=.*[A-Z])/) //pelo menos uma maiuscula
                        .matches(/^(?=.*[a-z])/) // pelo menos uma minuscula
                        .matches(/^(?=.*[0-9])/) // pelo menos um nuemro
                        .isValidSync(text)
                    );
                  }}
                />
                <Box mt="micro" mb="quarck">
                  <TouchableOpacity
                    onPress={() => {
                      navigation.navigate('ForgotEmail', {});
                    }}
                  >
                    <Typography style={{ textDecorationLine: 'underline' }}>
                      Esqueci minha senha
                    </Typography>
                  </TouchableOpacity>
                </Box>
                {loginCredentials.hasError && (
                  <Typography
                    color="vermelhoAlerta"
                    fontFamily="nunitoRegular"
                    fontSize={13}
                  >
                    {loginCredentials.showMessageError}
                  </Typography>
                )}
              </Box>
            )}
          </Box>
          <Box mt="md" />
          <Button
            title={!loginWithCode ? 'ENTRAR' : 'RECEBER CÓDIGO'}
            inline
            variant="primarioEstreito"
            disabled={loadingSendMail || loading}
            onPress={() => (loginWithCode ? handleLoginCode() : handleLogin())}
          />
          <Box my={50}>
            <Typography variant="tituloSessao" textAlign="center">
              OU
            </Typography>
          </Box>
          <Button
            title={
              loginWithCode
                ? 'ENTRAR COM LOGIN E SENHA'
                : 'RECEBER CÓDIGO DE ACESSO'
            }
            inline
            variant="primarioEstreitoOutline"
            onPress={() => {
              setLoginWithCode(!loginWithCode)

              //remove a mensagem de erro do campo email
              removeMessageErrorEmail()
            }}
          />
          {/* <Box
            flexDirection="row"
            mt="sm"
            justifyContent="center"
            alignItems="center"
          >
            <Typography textAlign="center">
              {"Ainda não possui uma conta?"}
            </Typography>
            <Button>
              <Typography style={{ textDecorationLine: "underline" }}>
                Clique para se cadastrar
              </Typography>
            </Button>
          </Box> */}
        </Box>
      </ScrollView>
    </SafeAreaView>
  );
};
