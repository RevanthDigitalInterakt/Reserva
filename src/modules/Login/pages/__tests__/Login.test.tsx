import React from 'react';

import { render, screen } from '@testing-library/react-native';
import type { StackScreenProps } from '@react-navigation/stack';
import { MockedProvider } from '@apollo/client/testing/react/MockedProvider';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@usereservaapp/reserva-ui';
import { LoginScreen } from '../Login';
import type { RootStackParamList } from '../../../../routes/StackNavigator';
import CartContextProvider from '../../../../context/CartContext';

type TNavigation = StackScreenProps<RootStackParamList, 'LoginAlternative'>['navigation'];

const mockedNavigate = jest.fn();
const mockGoBackFn = jest.fn();
const mockAddListenerFn = jest.fn();

const navigationMock: Partial<TNavigation> = {
  navigate: mockedNavigate,
  goBack: mockGoBackFn,
  addListener: mockAddListenerFn,
};

const Component = (

  <ThemeProvider theme={theme}>
    <MockedProvider addTypename={false}>
      <CartContextProvider>
        <LoginScreen
          navigation={navigationMock as TNavigation}
          route={{
            name: 'LoginAlternative',
            key: '',
            params: '',
          }}
        />
      </CartContextProvider>
    </MockedProvider>
  </ThemeProvider>
);

describe('Login', () => {
  it('snapshot', () => {
    render(Component);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
