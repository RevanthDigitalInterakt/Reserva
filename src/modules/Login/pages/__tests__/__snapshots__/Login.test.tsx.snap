// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      left={10}
      position="absolute"
      style={
        Array [
          Object {
            "left": 10,
            "position": "absolute",
            "top": 10,
            "zIndex": 4,
          },
          Object {
            "elevation": 3,
          },
        ]
      }
      top={10}
      zIndex={4}
    >
      <View
        alignItems="center"
        style={
          Array [
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "opacity": 1,
            }
          }
        >
          <RNSVGSvgView
            bbHeight={24}
            bbWidth={24}
            color="preto"
            focusable={false}
            size={24}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Array [
                  Object {
                    "color": "#333333",
                    "height": 24,
                    "width": 24,
                  },
                ],
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <RNSVGGroup>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="100%"
                bbWidth="100%"
                focusable={false}
                height="100%"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    Array [
                      Object {
                        "height": "100%",
                        "width": "100%",
                      },
                    ],
                    Object {
                      "flex": 0,
                      "height": "100%",
                      "width": "100%",
                    },
                  ]
                }
                vbHeight={19.054}
                vbWidth={18.225}
                width="100%"
              >
                <RNSVGGroup>
                  <RNSVGGroup
                    matrix={
                      Array [
                        -1,
                        1.2246467991473532e-16,
                        -1.2246467991473532e-16,
                        -1,
                        17.225,
                        17.639,
                      ]
                    }
                    name="Seta_-_icon"
                  >
                    <RNSVGPath
                      d="M7.5,18H23.725"
                      fill={null}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          -7.5,
                          -9.887,
                        ]
                      }
                      name="Caminho_68"
                      propList={
                        Array [
                          "fill",
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={4281545523}
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                    <RNSVGPath
                      d="M18,7.5l8.113,8.113L18,23.725"
                      fill={null}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          -9.887,
                          -7.5,
                        ]
                      }
                      name="Caminho_69"
                      propList={
                        Array [
                          "fill",
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      stroke={4281545523}
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <Image
      onError={[Function]}
      resizeMode="cover"
      source={
        Object {
          "commons": Object {
            "GuideBermuda": [Circular],
            "GuideCalca": [Circular],
            "GuideCamiseta": [Circular],
            "GuideHoodie": [Circular],
            "GuideJaqueta": [Circular],
            "GuideMangaCurta": [Circular],
            "GuideMangaLonga": [Circular],
            "GuidePolo": [Circular],
            "GuideShort": [Circular],
            "GuideSueter": [Circular],
            "GuideSungaCueca": [Circular],
            "QrcodeBackground": [Circular],
            "americanExpress": [Circular],
            "arrowInstruction": [Circular],
            "bagEmpty": [Circular],
            "bannerCatalog": [Circular],
            "bannerLogin": [Circular],
            "bannerOffer": [Circular],
            "cardFront": [Circular],
            "cardVerso": [Circular],
            "clothingCare1": [Circular],
            "clothingCareBleach": [Circular],
            "clothingCareDrying": [Circular],
            "clothingCarePass": [Circular],
            "clothingCareProfessionalWashing": [Circular],
            "clothingCareWash": [Circular],
            "cupomModalBackground": [Circular],
            "dividerReserva": [Circular],
            "elo": [Circular],
            "emptyProductCatalog": [Circular],
            "facebook": [Circular],
            "foraDoAr": [Circular],
            "headerLogin": [Circular],
            "homeMock1": [Circular],
            "homeMock2": [Circular],
            "homeMock3": [Circular],
            "homeMock4": [Circular],
            "imageCashback": [Circular],
            "imageNotFound": [Circular],
            "instagram": [Circular],
            "localReserva": [Circular],
            "logoQRCode": [Circular],
            "mastercard": [Circular],
            "noStoresFound": [Circular],
            "noWishList": [Circular],
            "pinYou": [Circular],
            "purchaseConfirmation": [Circular],
            "raceImageBackground": [Circular],
            "saleOff": [Circular],
            "searchNotFound": [Circular],
            "selectRectangle": [Circular],
            "shirt1": [Circular],
            "shirt2": [Circular],
            "shirt3": [Circular],
            "shirt4": [Circular],
            "topBannerMock1": [Circular],
            "twitter": [Circular],
            "update": [Circular],
            "visa": [Circular],
            "whatsapp": [Circular],
            "withoutInternet": [Circular],
            "zoomHand": [Circular],
          },
          "default": Object {
            "GuideBermuda": [Circular],
            "GuideCalca": [Circular],
            "GuideCamiseta": [Circular],
            "GuideHoodie": [Circular],
            "GuideJaqueta": [Circular],
            "GuideMangaCurta": [Circular],
            "GuideMangaLonga": [Circular],
            "GuidePolo": [Circular],
            "GuideShort": [Circular],
            "GuideSueter": [Circular],
            "GuideSungaCueca": [Circular],
            "QrcodeBackground": [Circular],
            "americanExpress": [Circular],
            "arrowInstruction": [Circular],
            "bagEmpty": [Circular],
            "bannerCatalog": [Circular],
            "bannerLogin": [Circular],
            "bannerOffer": [Circular],
            "cardFront": [Circular],
            "cardVerso": [Circular],
            "chevronRight": [Function],
            "clothingCare1": [Circular],
            "clothingCareBleach": [Circular],
            "clothingCareDrying": [Circular],
            "clothingCarePass": [Circular],
            "clothingCareProfessionalWashing": [Circular],
            "clothingCareWash": [Circular],
            "cupomModalBackground": [Circular],
            "dividerReserva": [Circular],
            "elo": [Circular],
            "emptyProductCatalog": [Circular],
            "facebook": [Circular],
            "foraDoAr": [Circular],
            "headerLogin": [Circular],
            "homeMock1": [Circular],
            "homeMock2": [Circular],
            "homeMock3": [Circular],
            "homeMock4": [Circular],
            "imageCashback": [Circular],
            "imageNotFound": [Circular],
            "instagram": [Circular],
            "localReserva": [Circular],
            "logoPrime": [Function],
            "logoQRCode": [Circular],
            "mastercard": [Circular],
            "noStoresFound": [Circular],
            "noWishList": [Circular],
            "pinYou": [Circular],
            "purchaseConfirmation": [Circular],
            "raceImageBackground": [Circular],
            "saleOff": [Circular],
            "searchNotFound": [Circular],
            "selectRectangle": [Circular],
            "shirt1": [Circular],
            "shirt2": [Circular],
            "shirt3": [Circular],
            "shirt4": [Circular],
            "topBannerMock1": [Circular],
            "twitter": [Circular],
            "update": [Circular],
            "visa": [Circular],
            "whatsapp": [Circular],
            "withoutInternet": [Circular],
            "zoomHand": [Circular],
          },
          "svgs": Object {
            "chevronRight": [Function],
            "logoPrime": [Function],
          },
        }
      }
      style={
        Array [
          Object {
            "height": 0,
            "width": 750,
          },
          undefined,
        ]
      }
    />
  </View>
  <RCTScrollView
    accessibilityLabel="com.usereserva:id/login_scrollview"
    keyboardShouldPersistTaps="always"
    testID="com.usereserva:id/login_scrollview"
  >
    <View>
      <View
        paddingBottom="xxl"
        pt="xxs"
        px="xxs"
        style={
          Array [
            Object {
              "paddingBottom": 80,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
            },
          ]
        }
      >
        <Text
          fontFamily="reservaSerifRegular"
          fontSize={22}
          style={
            Array [
              Object {
                "fontFamily": "ReservaSerif-Regular",
                "fontSize": 22,
              },
            ]
          }
        >
          Boas-vindas!
        </Text>
        <View
          mt="xxs"
          style={
            Array [
              Object {
                "marginTop": 24,
              },
            ]
          }
        >
          <View
            marginBottom="xxxs"
            style={
              Array [
                Object {
                  "marginBottom": 16,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "Nunito-Regular",
                    "fontSize": 15,
                  },
                ]
              }
              variant="tituloSessao"
            >
              Insira seu e-mail para continuar:
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "width": 710,
                },
              ]
            }
            width={710}
          >
            <View
              borderBottomColor="neutroFrio2"
              borderBottomWidth="hairline"
              flexDirection="row"
              justifyContent="space-between"
              style={
                Array [
                  Object {
                    "borderBottomColor": "#8A8C8E",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  },
                  Object {
                    "overflow": "hidden",
                  },
                ]
              }
            >
              <View
                flexGrow={4}
                style={
                  Array [
                    Object {
                      "flexGrow": 4,
                    },
                  ]
                }
              >
                <TextInput
                  accessibilityLabel="com.usereserva:id/login_input_email"
                  allowFontScaling={true}
                  autoCapitalize="none"
                  autoCompleteType="off"
                  autoCorrect={false}
                  collapsable={false}
                  keyboardType="email-address"
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  placeholder="Digite seu e-mail"
                  rejectResponderTermination={true}
                  secureTextEntry={false}
                  style={
                    Object {
                      "margin": 0,
                      "maxWidth": 710,
                      "padding": 0,
                      "paddingRight": 0,
                    }
                  }
                  testID="com.usereserva:id/login_input_email"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <View
            mt="md"
            style={
              Array [
                Object {
                  "marginTop": 48,
                  "width": "100%",
                },
              ]
            }
            width="100%"
          >
            <View
              style={
                Array [
                  Object {
                    "width": 710,
                  },
                ]
              }
              width={710}
            >
              <View
                borderBottomColor="neutroFrio2"
                borderBottomWidth="hairline"
                flexDirection="row"
                justifyContent="space-between"
                style={
                  Array [
                    Object {
                      "borderBottomColor": "#8A8C8E",
                      "borderBottomWidth": 1,
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    },
                    Object {
                      "overflow": "hidden",
                    },
                  ]
                }
              >
                <View
                  flexGrow={4}
                  style={
                    Array [
                      Object {
                        "flexGrow": 4,
                      },
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="com.usereserva:id/login_input_password"
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    collapsable={false}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    onGestureHandlerEvent={[Function]}
                    onGestureHandlerStateChange={[Function]}
                    placeholder="Digite sua senha"
                    rejectResponderTermination={true}
                    secureTextEntry={true}
                    style={
                      Object {
                        "margin": 0,
                        "maxWidth": 684,
                        "padding": 0,
                        "paddingRight": 0,
                      }
                    }
                    testID="com.usereserva:id/login_input_password"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "right": 14,
                    }
                  }
                >
                  <View
                    alignItems="center"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                        },
                      ]
                    }
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "opacity": 1,
                        }
                      }
                      testID="com.usereserva:id/login_input_password_button_hide_password"
                    >
                      <RNSVGSvgView
                        bbHeight={22}
                        bbWidth={22}
                        color="preto"
                        focusable={false}
                        size={22}
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "borderWidth": 0,
                            },
                            Array [
                              Object {
                                "color": "#333333",
                                "height": 22,
                                "width": 22,
                              },
                            ],
                            Object {
                              "flex": 0,
                              "height": 22,
                              "width": 22,
                            },
                          ]
                        }
                      >
                        <RNSVGGroup>
                          <RNSVGSvgView
                            align="xMidYMid"
                            bbHeight="100%"
                            bbWidth="100%"
                            fill="none"
                            focusable={false}
                            height="100%"
                            meetOrSlice={0}
                            minX={0}
                            minY={0}
                            stroke="#333333"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                            strokeWidth={2}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "borderWidth": 0,
                                },
                                Array [
                                  Object {
                                    "height": "100%",
                                    "width": "100%",
                                  },
                                ],
                                Object {
                                  "flex": 0,
                                  "height": "100%",
                                  "width": "100%",
                                },
                              ]
                            }
                            vbHeight={24}
                            vbWidth={24}
                            width="100%"
                          >
                            <RNSVGGroup
                              fill={null}
                              propList={
                                Array [
                                  "fill",
                                  "stroke",
                                  "strokeWidth",
                                  "strokeLinecap",
                                  "strokeLinejoin",
                                ]
                              }
                              stroke={4281545523}
                              strokeLinecap={1}
                              strokeLinejoin={1}
                              strokeWidth={2}
                            >
                              <RNSVGPath
                                d="M17.94 17.94A10.07 10.07 0 0112 20c-7 0-11-8-11-8a18.45 18.45 0 015.06-5.94M9.9 4.24A9.12 9.12 0 0112 4c7 0 11 8 11 8a18.5 18.5 0 01-2.16 3.19m-6.72-1.07a3 3 0 11-4.24-4.24"
                              />
                              <RNSVGPath
                                d="M1 1L23 23"
                              />
                            </RNSVGGroup>
                          </RNSVGSvgView>
                        </RNSVGGroup>
                      </RNSVGSvgView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              mb="quarck"
              mt="micro"
              style={
                Array [
                  Object {
                    "marginBottom": 4,
                    "marginTop": 12,
                  },
                ]
              }
            >
              <RNGestureHandlerButton
                collapsable={false}
                onGestureEvent={[Function]}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                onHandlerStateChange={[Function]}
                rippleColor={0}
              >
                <View
                  accessible={true}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    accessibilityLabel="com.usereserva:id/esqueci-minha-senha"
                    style={
                      Array [
                        Object {},
                        Object {
                          "textDecorationLine": "underline",
                        },
                      ]
                    }
                    testID="com.usereserva:id/esqueci-minha-senha"
                  >
                    Esqueci minha senha
                  </Text>
                </View>
              </RNGestureHandlerButton>
            </View>
          </View>
        </View>
        <View
          mt="md"
          style={
            Array [
              Object {
                "marginTop": 48,
              },
            ]
          }
        />
        <View
          alignItems="stretch"
          style={
            Array [
              Object {
                "alignItems": "stretch",
              },
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/entrar_login_button"
            accessible={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#333333",
                "borderWidth": 1,
                "color": "#333333",
                "fontFamily": "Nunito-Bold",
                "fontSize": 13,
                "justifyContent": "center",
                "minWidth": 130,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 16,
              }
            }
            testID="com.usereserva:id/entrar_login_button"
          >
            <View
              alignItems="center"
              flexDirection="row"
              justifyContent="center"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  color="preto"
                  fontFamily="nunitoBold"
                  fontSize={13}
                  letterSpacing={1.6}
                  style={
                    Array [
                      Object {
                        "color": "#333333",
                        "fontFamily": "Nunito-Bold",
                        "fontSize": 13,
                        "letterSpacing": 1.6,
                      },
                    ]
                  }
                >
                  ENTRAR
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          alignItems="center"
          flexDirection="row"
          justifyContent="center"
          mb="xxs"
          mt="xxl"
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 24,
                "marginTop": 80,
              },
            ]
          }
        >
          <View
            borderColor="divider"
            flex={1}
            marginLeft="xxs"
            marginRight="nano"
            style={
              Array [
                Object {
                  "borderColor": "#D8D9DA",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 24,
                  "marginRight": 8,
                },
                Object {
                  "borderWidth": 1,
                },
              ]
            }
          />
          <Text
            style={
              Array [
                Object {
                  "textAlign": "center",
                },
              ]
            }
            textAlign="center"
          >
            Ainda não possui uma conta?
          </Text>
          <View
            borderColor="divider"
            flex={1}
            marginLeft="nano"
            marginRight="xxs"
            style={
              Array [
                Object {
                  "borderColor": "#D8D9DA",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "marginLeft": 8,
                  "marginRight": 24,
                },
                Object {
                  "borderWidth": 1,
                },
              ]
            }
          />
        </View>
        <View
          alignItems="stretch"
          style={
            Array [
              Object {
                "alignItems": "stretch",
              },
            ]
          }
        >
          <View
            accessibilityLabel="com.usereserva:id/cadastre_se_buttton"
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#333333",
                "color": "#FFF",
                "fontFamily": "Nunito-Bold",
                "fontSize": 13,
                "justifyContent": "center",
                "minWidth": 130,
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 16,
              }
            }
            testID="com.usereserva:id/cadastre_se_buttton"
          >
            <View
              alignItems="center"
              flexDirection="row"
              justifyContent="center"
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <View
                justifyContent="center"
                style={
                  Array [
                    Object {
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Text
                  color="white"
                  fontFamily="nunitoBold"
                  fontSize={13}
                  letterSpacing={1.6}
                  style={
                    Array [
                      Object {
                        "color": "#FFF",
                        "fontFamily": "Nunito-Bold",
                        "fontSize": 13,
                        "letterSpacing": 1.6,
                      },
                    ]
                  }
                >
                  CADASTRE-SE
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
